src/controllers/tripRequestController.js

```diff
@@ -0,0 +1,205 @@
+            if (!trip) {
+                return res.status(404).json({ success: false, message: ` Trip Request with id= ${requestId} Not Found` })
+            } else {
+                return res.status(200).json({ success: true, data: trip })
+            }
+        } else {
+            return res.status(403).json({ success: false, message: 'UnAuthorized' })
+        }
+    } catch (error) {
+        res.status(500).json({ message: 'error while getting trip request', error: error.message })
+    }
+};
+
+export const getAllTripRequest = async (req, res) => {
+    try {
+        const userId = req.user.id;
+
+        if (req.user.role == 'requester') {
+            const trips = await tripRequests.findAll({
+                where: { requesterId: userId },
+                include: [{ model: accomodations, as: 'accomodation', attributes: { exclude: ['createdAt', 'updatedAt'] } }],
+                attributes: { exclude: ['createdAt', 'updatedAt', 'accomodationId'] }
+            })
+
+            if (!trips) {
+                return res.status(404).json({ message: `there are no Trip Requests assigned to ${userId}` })
+            } else {
+                return res.status(200).json({ success: true, data: trips })
+            }
+
+        } else if (req.user.role == 'manager') {
+            const trips = await tripRequests.findAll({
+                include: [{ model: accomodations, as: 'accomodation', attributes: { exclude: ['createdAt', 'updatedAt'] } }],
+                attributes: { exclude: ['createdAt', 'updatedAt', 'accomodationId'] }
+            })
+
+            if (!trips) {
+                return res.status(404).json({ success: false, message: `No Trip Requests found` })
+            } else {
+                return res.status(200).json({ success: true, data: trips })
+            }
+        } else {
+            return res.status(403).json({ message: 'Unauthorized' })
+        }
+    } catch (error) {
+        res.status(500).json({ message: 'error while getting trip request', error: error.message })
+    }
+}
+
+export const updateTripRequest = async (req, res) => {
+    try {
+        if (req.user.role !== 'requester') {
+            return res.status(403).json({ message: 'Unauthorized' })
+        }
+        const requestId = req.params.id;
+        const userId = req.user.id;
+
+        const tripRequest = await tripRequests.findOne({
+            where: { id: requestId }
+        })
+
+
+        if (!tripRequest) {
+            return res.status(404).json({ status: 'fail', message: `Trip Request with id = ${requestId} Not Found!` })
+        }
+
+        if (tripRequest.status == 'pending') {
+
+            await tripRequestSchema.validateAsync(req.body);
+            const accomodation = await accomodations.findOne({ where: { id: req.body.accomodationId } })
```

Comment:
there is no need of checking if accomodation exist, in you updated request you do not have to send the accommodation id again
 `"leavingFrom": "kigali",
  "goingTo": 5,
  "travelDate": "2022-10-5",
  "returnDate": "2022-10-10",
  "travelReason": "leasure",`
just leave it and update as usual